js/source/raster_tile_source.js

```diff
@@ -66,6 +66,9 @@ RasterTileSource.prototype = util.inherit(Evented, {
                 gl.texSubImage2D(gl.TEXTURE_2D, 0, 0, 0, gl.RGBA, gl.UNSIGNED_BYTE, img);
             } else {
                 tile.texture = gl.createTexture();
+                if (!tile.texture) {
+                    return callback(null);
+                }
```

Comment:
> The first npe occurs because gl.createTexture() can silently return null if the webgl context has been closed.

I would have thought we were aborting all outstanding tile requests during a `Map#remove` operation but that does not seem to be the case. I would go about fixing this bug by adding `source.clearTiles()` to `Style#removeSource`. 

Without a demonstration of the bug, it is impossible for me to know for sure.